momepy/coins.py

```diff
@@ -377,119 +381,59 @@ def _list_to_pairs(in_list):
     """
-    point1 = line[1]
-    point2 = line[0]
-
-    # If the latutide of a point is less and the longitude is more, or
-    # If the latitude of a point is more and the longitude is less, then
-    # the point is oriented leftward and wil have negative orientation.
-    if ((point2[0] > point1[0]) and (point2[1] < point1[1])) or (
-        (point2[0] < point1[0]) and (point2[1] > point1[1])
-    ):
-        return -_compute_angle(point1, point2)
-
-    # if the latitudes are same, the line is horizontal
-    elif point2[1] == point1[1]:
-        return 0
-
-    # if the longitudes are same, the line is vertical
-    elif point2[0] == point1[0]:
-        return 90
-
-    return _compute_angle(point1, point2)
-
+    computing orientation is obsolete, but:
+    keeping it in for now not to mess up with the
+    index-based logic
+    (required by _split_lines() )
+    """
+    return None
 
-def _points_set_angle(line1, line2):
-    """Calculate the acute joining angle between two given set of points."""
-    l1orien = _compute_orientation(line1)
-    l2orien = _compute_orientation(line2)
+def _angle_between_two_lines(line1, line2):
+    """
+    Computes interior angle between 2 lines.
+        input:  line<x> ... list of 2 tuples (x,y);
+                line1 and line2 by definition share one unique tuple
+                (overlap in 1 point)
+        returns:    interior angle in degrees 0<alpha<=180
+                    (we assume that line1!=line2, so alpha=0 not possible)
+    """
 
-    if ((l1orien > 0) and (l2orien < 0)) or ((l1orien < 0) and (l2orien > 0)):
-        return abs(l1orien) + abs(l2orien)
+    # extract points
+    a, b = tuple(line1[0]), tuple(line1[1])
+    c, d = tuple(line2[0]), tuple(line2[1])
 
-    elif ((l1orien > 0) and (l2orien > 0)) or ((l1orien < 0) and (l2orien < 0)):
-        theta1 = abs(l1orien) + 180 - abs(l2orien)
-        theta2 = abs(l2orien) + 180 - abs(l1orien)
-        if theta1 < theta2:
-            return theta1
-        else:
-            return theta2
+    # assertion: we expect exactly 2 of the 4 points to be identical
+    # (lines touch in this point)
+    points = Counter([a, b, c, d])
 
-    elif (l1orien == 0) or (l2orien == 0):
-        if l1orien < 0:
-            return 180 - abs(l1orien)
-        elif l2orien < 0:
-            return 180 - abs(l2orien)
-        else:
-            return 180 - (
-                abs(_compute_orientation(line1)) + abs(_compute_orientation(line2))
-            )
+    # make sure lines are not identical
+    if Counter(points.values()) == {2: 2}:
+        raise ValueError(
+            "ValueError: lines are identical. Please revise input gdf\
+            to ensure no lines are identical or overlapping."
+        )
 
-    elif l1orien == l2orien:
-        return 180
+    # make sure lines do touch
+    if Counter(points.values()) == {1: 4}:
+        raise ValueError("ValueError: lines do not touch.")
```

Comment:
```suggestion
        raise ValueError("Lines do not touch.")
```

No need to write "ValueError" in the message. That would result in "**ValueError:** ValueError: message"